python/hsml/core/explicit_provenance.py

```diff
@@ -0,0 +1,338 @@
+            "name": self._name,
+            "version": self._version,
+        }
+
+    def __repr__(self):
+        return (
+            f"Artifact({self._model_registry_id!r}, {self._name!r}, "
+            f"{self._version!r}, {self._type!r}, {self._meta_type!r}, "
+            f"{self._href!r}, {self._exception_cause!r})"
+        )
+
+    @staticmethod
+    def from_response_json(json_dict: dict):
+        link_json = humps.decamelize(json_dict)
+        if link_json.get("exception_cause") is not None:
+            return Artifact(
+                link_json["artifact"]["project"],
+                link_json["artifact"]["name"],
+                link_json["artifact"]["version"],
+                link_json["artifact_type"],
+                Artifact.MetaType.FAULTY,
+            )
+        elif bool(link_json["deleted"]):
+            return Artifact(
+                link_json["artifact"]["project"],
+                link_json["artifact"]["name"],
+                link_json["artifact"]["version"],
+                link_json["artifact_type"],
+                Artifact.MetaType.DELETED,
+            )
+        elif not bool(link_json["accessible"]):
+            return Artifact(
+                link_json["artifact"]["project"],
+                link_json["artifact"]["name"],
+                link_json["artifact"]["version"],
+                link_json["artifact_type"],
+                Artifact.MetaType.INACCESSIBLE,
+                link_json["artifact"]["href"],
+            )
+        elif bool(link_json["accessible"]):
+            return Artifact(
+                link_json["artifact"]["project"],
+                link_json["artifact"]["name"],
+                link_json["artifact"]["version"],
+                link_json["artifact_type"],
+                Artifact.MetaType.NOT_SUPPORTED,
+                link_json["artifact"]["href"],
+            )
+        else:
+            return None
+
+
+class Links:
+    def __init__(self):
+        self._accessible = []
+        self._deleted = []
+        self._inaccessible = []
+        self._faulty = []
+
+    @property
+    def deleted(self):
+        """List of [Artifact objects](../links#artifact) which contains
+        minimal information (name, version) about the entities
+        (feature groups, feature views) they represent.
+        These entities have been removed from the feature store.
+        """
+        return self._deleted
+
+    @property
+    def inaccessible(self):
+        """List of [Artifact objects](../links#artifact) which contains
+        minimal information (name, version) about the entities
+        (feature groups, feature views) they represent.
+        These entities exist in the feature store, however the user
+        does not have access to them anymore.
+        """
+        return self._inaccessible
+
+    @property
+    def accessible(self):
+        """List of [feature groups](../feature_group_api) or
```

Comment:
Are the links valid?

Reply:
removed... pretty sure these don't work cross libraries, so i removed all